eva-accession-ws/src/main/java/uk/ac/ebi/eva/accession/ws/rest/ClusteredVariantsRestController.java

```diff
@@ -49,5 +63,30 @@ public ClusteredVariantsRestController(
             @PathVariable List<Long> identifiers) {
         return basicRestController.get(identifiers);
     }
+
+    @ApiOperation(value = "Find submitted variants by clustered variant identifier")
+    @GetMapping(value = "/{identifiers}/submitted", produces = "application/json")
+    public List<AccessionResponseDTO<SubmittedVariant, ISubmittedVariant, String, Long>> getSubmittedVariants(
+            @PathVariable List<Long> identifiers) {
+        List<DbsnpSubmittedVariantEntity> submittedVariantEntities = submittedVariantsRepository
+                .findByClusteredVariantAccessionIn(identifiers);
+
+        // return retrieved SubmittedVariant objects encapsulated in AccessionResponseDTO objects
+        return submittedVariantEntities.stream().map(this::submittedVariantToAccessionWrapper).map(
+                this::acessionWrapperToResponseDTO).collect(Collectors.toList());
+    }
+
+    private AccessionWrapper<ISubmittedVariant, String, Long> submittedVariantToAccessionWrapper(
```

Comment:
I don't like repeating these convertion methods, but also don't see a good way to reuse them, so :man_shrugging:  :+1:

Reply:
Instead of directly accessing the repository, couldn't we extend the interfaces of service classes? That would keep the repository in the layer it should be (service instead of API), as well as this transformation.